core/labeled-replies-3.3.md

```diff
@@ -0,0 +1,132 @@
+
+## Architecture
+
+### Dependencies
+
+This specification uses the [message tags framework](/specs/core/message-tags-3.2.html) and relies on support for the [`batch`](/specs/extensions/batch-3.2.html) capability which MUST be negotiated at the same time.
+
+### Capabilities
+
+This speciciation adds the `draft/labeled-response` capability.
+
+Clients requesting this capability indicate that they are capable of handling the message tag described below from servers.
+
+Servers advertising this capability indicate that they are capable of handling the message tag described below from clients, and will use with the same tag and value in their response.
+
+Servers MUST reject the capability at negotiation time if the `batch` capability is not also negotiated. Additionally, servers supporting `cap-notify` MUST cancel the capability if `batch` is removed at any time.
+
+### Batch types
+
+This specification adds the `draft/labeled-response` batch type, described below.
+
+### Tags
+
+This specification adds the `draft/label` message tag, which has one required value.
+
+This tag MAY be sent by a client for any messages that need to be correlated with a response from the server.
+
+For any message received from a client that includes this tag, the server MUST include the same tag and value in any response required from this message. Servers MUST include the tag in exactly one message.
+
+If a response conists of more than one message, a `batch` MUST be used to group them into a single logical response. The start of the batch MUST be tagged with the `draft/label` tag. The batch type MUST be one of:
+
+* An existing type applicable to the entire response
+* `draft/labeled-response`
+
+If no response is required, an empty batch MUST be sent.
+
+#### Tag value
+
+The tag value is chosen by the client and MUST be treated as an opaque identifier. The client SHOULD NOT reuse a tag value until it has received a complete response for that value from the server.
+
+## Client implementation considerations
+
+This section is non-normative.
+
+In the case of `echo-message` (see example below), a client can use labeled responses to correlate a server's acknowledgment of their own messages with a temporary message displayed locally. The temporary message can be displayed to the user immediately in a pending state to reduce perceived lag, and then removed once a labeled response from the server is received.
+
+When sending messages directed at a client's own nick, `echo-message` will result in duplicate messages being sent by the server, as both sent and received messages. Labeled responses allow clients to deduplicate these messages in one of two ways:
+
+For private messages that match the clients nickname:
+1. Ignore labeled messages, and use any unlabeled message as acknowledgment for all sent messages to clear temporary local messages.
```

Comment:
When this cap is used together with echo-message and the client messages itself then both messages (one from echo-message and the other one being the delivered message) will be labeled because both of those events were generated because of the labeled PRIVMSG the client sent to the server, i.e. both are "replies". This means "unlabeled messages" don't exist in these cases, or do I misunderstand something?